packages/panna-sdk/src/core/utils/utils.ts

```diff
@@ -207,3 +209,78 @@ export const accountBalanceInFiat = async function (
 
   return result;
 };
+
+/**
+ * Get the total fiat value of multiple tokens and individual token balances
+ * @param params - Parameters for getting multiple account balances
+ * @returns Total value and individual token balances with fiat values
+ * @throws Error if address is invalid or if any token fetch fails
+ * @example
+ * ```ts
+ * // Get portfolio value for multiple tokens
+ * const result = await accountBalancesInFiat({
+ *   client: pannaClient,
+ *   address: '0x...',
+ *   chain: lisk,
+ *   tokens: [
+ *     {}, // Native token
+ *     { address: '0x...' }, // USDC
+ *     { address: '0x...' }  // DAI
+ *   ],
+ *   currency: 'USD'
+ * });
+ * // result: {
+ * //   totalValue: { amount: 5250.75, currency: 'USD' },
+ * //   tokenBalances: [...]
+ * // }
+ * ```
+ */
+export async function accountBalancesInFiat(
+  params: AccountBalancesInFiatParams
+): Promise<AccountBalancesInFiatResult> {
+  if (!isValidAddress(params.address)) {
+    throw new Error('Invalid address format');
+  }
+
+  const currency = params.currency || 'USD';
+
+  // Create array of promises for parallel execution
+  const balancePromises = params.tokens.map((tokenInfo) => {
+    // Validate token address upfront
+    if (tokenInfo.address && !isValidAddress(tokenInfo.address)) {
+      throw new Error(`Invalid token address format: ${tokenInfo.address}`);
+    }
+
+    // Return promise with error handling
+    return accountBalanceInFiat({
+      address: params.address,
+      client: params.client,
+      chain: params.chain,
+      tokenAddress: tokenInfo.address,
+      currency: currency
+    }).catch((error) => {
+      // Re-throw with more context about which token failed
+      const tokenIdentifier = tokenInfo.address || 'native token';
+      throw new Error(
+        `Failed to get balance for ${tokenIdentifier}: ${error instanceof Error ? error.message : 'Unknown error'}`
+      );
+    });
+  });
+
+  // Execute all balance fetches in parallel
+  const tokenBalances = await Promise.all(balancePromises);
```

Comment:
Use `Promise.allSettled` instead to accommodate rejections as well. `Promise.all` only returns in case all the promises resolve.
```suggestion
  const tokenBalances = await Promise.allSettled(balancePromises);
```